File: chapter3/exercise_26/cpp/connect_with_separator.cpp
#include <iostream>
#include <iterator>
#include <sstream>
#include <vector>
#include <string>

template <typename Container>
std::string join_strings(const Container& c, const char* const separator)
{
    std::ostringstream os;
    std::copy(std::begin(c), std::end(c)-1, std::ostream_iterator<std::string>(os, separator));
    os << *(std::end(c)-1);
    return os.str();
}

int main()
{
    std::cout << join_strings(std::vector<std::string>{"one", "two", "three"}, ":elo:");
    return 0;
}

File: chapter3/exercise_26/go/connect_with_separator.go
package main

import (
	"fmt"
	"strings"
)

func join_string(strings []string, separator string) string {
	result := strings[0] 
	for idx := 1; idx < len(strings); idx++{
		result += separator + strings[idx] 
	}
	return result
}

func main() {
	fmt.Println(join_string([]string{"one", "two", "three"}, ":sep:"))
	fmt.Println(strings.Join([]string{"one", "two", "three"}, ":sep:"))
}
